core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -271,4 +318,59 @@ class HadoopMapReduceCommitProtocol(
         logWarning(s"Exception while aborting ${taskContext.getTaskAttemptID}", e)
     }
   }
+
+  /**
+   * Delete the staging output path of current InsertIntoHadoopFsRelation operation.
```

Comment:
This doc needs more detailed comments. Explain how we deleting staging dir and why we do that way.

Reply:
thanks